infrastructure/smart-contracts/contracts/RefereeCalculations.sol

```diff
@@ -21,32 +21,55 @@ contract RefereeCalculations is Initializable, AccessControlUpgradeable {
+     * 4. Multiply the emissions per hour by the time passed to get the total emissions for the challenge.
+     * 5. Return the total emissions and the emission tier.
      *
      * @param totalSupply The current total supply of tokens.
      * @param maxSupply The maximum supply of tokens.
-     * @return uint256 The challenge emission.
-     * @return uint256 The emission tier.
+     * @param challengeStart The start time of the challenge in Unix timestamp format.
+     * @param challengeEnd The end time of the challenge in Unix timestamp format.
+     * @return uint256 The total challenge emission based on the time passed.
+     * @return uint256 The emission tier for the current challenge.
      */
     function calculateChallengeEmissionAndTier(
         uint256 totalSupply,
-        uint256 maxSupply
+        uint256 maxSupply,
+        uint256 challengeStart,
+        uint256 challengeEnd
     ) public pure returns (uint256, uint256) {
         require(maxSupply > totalSupply, "5");
+        require(challengeEnd >= challengeStart, "7"); // Ensure that end time is not before start time
 
-        uint256 tier = Math.log2(maxSupply / (maxSupply - totalSupply)); // calculate which tier we are in starting from 0
-        require(tier <= 23, "6");
+        // 1. Calculate the time difference in seconds
+        uint256 timePassedInSeconds = challengeEnd - challengeStart;
 
-        uint256 emissionTier = maxSupply / (2 ** (tier + 1)); // equal to the amount of tokens that are emitted during this tier
+        // 2. Convert the time difference to fractional hours for precision
+        // We use 1e18 to maintain precision during division
+        uint256 timePassedInHours = (timePassedInSeconds * 1e18) / 3600;
 
-        // determine what the size of the emission is based on each challenge having an estimated static length
-        return (emissionTier / 17520, emissionTier);
+        // 3. Determine the current emission tier using a logarithmic approach
+        // The tier is calculated as the logarithm base 2 of the proportion of total supply to the remaining supply
+        uint256 tier = Math.log2(maxSupply / (maxSupply - totalSupply));
+        require(tier <= 23, "6"); // Limiting tier to a maximum value
+
+        // 4. Calculate the emission tier, which is the maximum supply divided by 2^(tier + 1)
+        uint256 emissionTier = maxSupply / (2 ** (tier + 1)); // Equal to the amount of tokens emitted during this tier
+
+        // 5. Calculate the emission rate per hour, based on the emission tier
+        uint256 emissionPerHour = emissionTier / 17520; // 17520 is the number of hours in a non-leap year
+
+        // 6. Calculate the total emissions based on the time passed, including fractional hours
+        // Multiply the emissions per hour by the time passed in fractional hours
+        uint256 totalEmissions = (emissionPerHour * timePassedInHours) / 1e18;
```

Comment:
Why are we not just calcualting the value of a second and then check the seconds that passed instead of scaling up by 18 decimals ?

Reply:
I was just dividing by the number of hours passed and since it could be less than 1, I scaled up, but I agree emissions per second makes more sense. 

I have revised to use emissions per second. If you see any other optimizations, please share. Thank you for the guidance!